### PR TITLE
[REFACTOR] `argilla`: Align questions to `Resource` API

### LINKED ISSUE #4931: [REFACTOR] Improve handling of question models and dicts
We can improve code quality and reduce redunancy by using a pydnatic type annotator instead of identical methods on each cleas:

@frascuchon gives an example here: https://github.com/argilla-io/argilla-python/pull/195#discussion_r1604995059

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

Closes https://github.com/argilla-io/argilla/issues/4931

**Type of change**

- Refactor (change restructuring the codebase without changing functionality)

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla/src/argilla/_api/_questions.py [hunk 1]
```diff
@@ -18,76 +18,56 @@
...
         self,
         question: QuestionModel,
     ) -> QuestionModel:
-        # TODO: Implement update method for fields with server side ID
-        raise NotImplementedError
+        url = f"/api/v1/questions/{question.id}"
+        response = self.http_client.patch(url, json=question.model_dump())
+        response.raise_for_status()
+        response_json = response.json()
+        updated_field = self._model_from_json(response_json)
```
Comment (burtenshaw) [inline argilla/src/argilla/_api/_questions.py]:
```suggestion
        updated_question = self._model_from_json(response_json)
```

**Review by burtenshaw — COMMENTED** [review]
This looks like a nice refactor and alignment. One question: does the new update API method allow updating questions? If so, how and when within the dataset lifecycle?

**Comment by frascuchon** [comment]
> This looks like a nice refactor and alignment. One question: does the new update API method allow updating questions? If so, how and when within the dataset lifecycle?

Yes. This change allows users to update some values for questions. It's true that currently there is not much flexibility to change questions (titles, descriptions, some flags) but this may change in the future (adding new labels for label or span questions,...). Fields have a similar flexibility degree and can be updated from the SDK.

File: argilla/src/argilla/_api/_questions.py [hunk 2]
```diff
@@ -18,76 +18,56 @@
...
+        self._log_message(message=f"Update question {updated_field.name} with id {question.id}")
+        return updated_field
```
Comment (frascuchon) [inline argilla/src/argilla/_api/_questions.py]:
```suggestion
        self._log_message(message=f"Update question {updated_question.name} with id {question.id}")
        return updated_question
```

**Review by burtenshaw — APPROVED** [review]
lgtm

**Pull request merged**